bgpd/bgp_zebra.c

```diff
@@ -1404,18 +1405,13 @@ void bgp_zebra_announce(struct bgp_dest *dest, const struct prefix *p,
 
 			if (!nexthop)
 				nh_updated = update_ipv4nh_for_route_install(
-					nh_othervrf,
-					nh_othervrf ? info->extra->bgp_orig
-						    : bgp,
+					nh_othervrf, bgp_orig,
 					&mpinfo_cp->attr->nexthop,
```

Comment:
those changes in bgp_zebra should also be on separate commit. this is just code optimisation, here.right ?

Reply:
Yes, that is correct.
one extra check of "mpinfo->sub_type == BGP_ROUTE_IMPORTED"
and imo more readable.

Please let us know, we can always avoid the change in bgp_zebra totally.

Reply:
Ack, will fix it.